src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.cs

```diff
@@ -639,6 +628,166 @@ save_availableCaptureId > (testExpressionCaptureId = ((IFlowCaptureReferenceOper
             return new FlowCaptureReference(resultCaptureId, operation.Syntax, operation.Type, operation.ConstantValue);
         }
 
+        private static bool IsNullableType(ITypeSymbol typeOpt)
+        {
+            return typeOpt?.OriginalDefinition.SpecialType == SpecialType.System_Nullable_T;
+        }
+
+        private static IOperation MakeInvalidOperation(ITypeSymbol type, IOperation child)
+        {
+            return new InvalidOperation(ImmutableArray.Create<IOperation>(child),
+                                        semanticModel: null, child.Syntax, type,
+                                        constantValue: default, isImplicit: true);
+        }
+
+        private static IOperation TryUnwrapNullableValue(IOperation value)
+        {
+            ITypeSymbol valueType = value.Type;
+
+            Debug.Assert(IsNullableType(valueType));
+
+            foreach (ISymbol candidate in valueType.GetMembers("GetValueOrDefault"))
+            {
+                if (candidate.Kind == SymbolKind.Method && !candidate.IsStatic && candidate.DeclaredAccessibility == Accessibility.Public)
+                {
+                    var method = (IMethodSymbol)candidate;
+                    if (method.Parameters.Length == 0 && !method.ReturnsByRef && !method.ReturnsByRefReadonly &&
+                        method.OriginalDefinition.ReturnType.Equals(((INamedTypeSymbol)valueType).OriginalDefinition.TypeParameters[0]))
```

Comment:
I really don't like this, but I can't see a good way to avoid re-implementing this logic either :-(
It seems error-prone (easy to forget when new features are added). Also seems like a bunch of cases that will need testing.